lib/era_ja/conversion.rb

```diff
@@ -21,31 +26,38 @@ def to_era(format = "%o%E.%m.%d")
       @era_format.sub!(/%E/) { |m| m + ' ' }
       str_time = strftime(@era_format)
       if @era_format =~ /%E/
-        if self.to_time < ::Time.mktime(1868,9,8)
-          raise "#to_era is expeted later in 1868,9,8"
-        elsif self.to_time < ::Time.mktime(1912,7,30)
-          str_time = era_year(year - 1867, 'M')
-        elsif self.to_time < ::Time.mktime(1926,12,25)
-          str_time = era_year(year - 1911, 'T')
-        elsif self.to_time < ::Time.mktime(1989,1,8)
-          str_time = era_year(year - 1925, 'S')
-        else
-          str_time = era_year(year - 1988, 'H')
-        end
+        str_time = era_year_str(*era_year)
       end
       str_time.gsub(/%J(\d+)/) { to_kanzi($1) }
     end
+    
+    # Get Japanese year.
+    # @return [String, Fixnum] (era_alphabet, era_num)
+    def era_year
+      checked_era = nil
+      checked_era_start = nil
+      ERA.each_pair do |key, arr|
```

Comment:
`each_pair` and `break` is not sense.

I recommend you use `detect`

If you move logic of checking invalid time and apply this comment, `detect` make our code to simple.

```
-      checked_era = nil
-      checked_era_start = nil
-      ERA.to_a.each do |key, arr|
-        _ , start = arr
-        break if self.to_time < start
-        checked_era = key
-        checked_era_start = start
-      end
-      [checked_era, year - checked_era_start.year + 1]
+      era = ERA.to_a.reverse.detect { |key, arr| self.to_time >= arr[1] }
+      [era[0], year - era[1][1].year + 1]
```